### PR TITLE
chore: change basepath

### PR DESCRIPTION
**Pull request by Skidaddle99**
Added a couple of DirectoryInfo extension methods, these help getting rid of all the "new DirectoryInfo(Path.Combine())" in most of the installation steps.

sc-18587

### CODE REVIEW

**Review by Victor-Allan — COMMENTED** [review]
Have you tested one of the steps to make sure? I know it should work just fine, but just in case

File: src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs [hunk 1]
```diff
@@ -14,100 +15,88 @@ public class ComposeMoveSitecoreConfigIncludes : AggregateStep
...
-                new DirectoryInfo(Path.Combine(sitecoreDirectory.FullName, "sitecore modules", "Shell", "ucommerce", "install", "configInclude"));
-            var appIncludeDirectory = new DirectoryInfo(Path.Combine(sitecoreDirectory.FullName, "App_Config", "include"));
+                sitecoreDirectory.CombineDirectory("sitecore modules", "Shell", "ucommerce", "install", "configInclude");
+            var appIncludeDirectory = sitecoreDirectory.CombineDirectory("App_Config", "include");
 
             Steps.AddRange(new IStep[]
             {
-                new MoveFile(new FileInfo(Path.Combine(configIncludeDirectory.FullName, "Sitecore.uCommerce.Databases.config")),
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Databases.config")),
+                new MoveFile(sitecoreDirectory.CombineFile("Sitecore.uCommerce.Databases.config"),
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs]: Should prolly be configIncludeDirectory

File: src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs [hunk 2]
```diff
@@ -14,100 +15,88 @@ public class ComposeMoveSitecoreConfigIncludes : AggregateStep
...
+                    appIncludeDirectory.CombineFile("Sitecore.uCommerce.Databases.config"),
                     true,
                     _loggingService),
-                new MoveFile(new FileInfo(Path.Combine(configIncludeDirectory.FullName, "Sitecore.uCommerce.Dataproviders.config")),
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Dataproviders.config")),
+                new MoveFile(sitecoreDirectory.CombineFile("Sitecore.uCommerce.Dataproviders.config"),
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs]: Should prolly be configIncludeDirectory

File: src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs [hunk 3]
```diff
@@ -14,100 +15,88 @@ public class ComposeMoveSitecoreConfigIncludes : AggregateStep
...
+                    appIncludeDirectory.CombineFile("Sitecore.uCommerce.Dataproviders.config"),
                     true,
                     _loggingService),
-                new MoveFile(new FileInfo(Path.Combine(configIncludeDirectory.FullName, "Sitecore.uCommerce.Events.config")),
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Events.config")),
+                new MoveFile(sitecoreDirectory.CombineFile("Sitecore.uCommerce.Events.config"),
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs]: Should prolly be configIncludeDirectory

File: src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs [hunk 4]
```diff
@@ -14,100 +15,88 @@ public class ComposeMoveSitecoreConfigIncludes : AggregateStep
...
+                    appIncludeDirectory.CombineFile("Sitecore.uCommerce.Events.config"),
                     true,
                     _loggingService),
-                new MoveFile(new FileInfo(Path.Combine(configIncludeDirectory.FullName, "Sitecore.uCommerce.Sites.config")),
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Sites.config")),
+                new MoveFile(sitecoreDirectory.CombineFile("Sitecore.uCommerce.Sites.config"),
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs]: Should prolly be configIncludeDirectory

File: src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs [hunk 5]
```diff
@@ -14,100 +15,88 @@ public class ComposeMoveSitecoreConfigIncludes : AggregateStep
...
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Pipelines.PreProcessRequest.config")),
+                new MoveFileIf(configIncludeDirectory.CombineFile(
+                        "Sitecore.uCommerce.Pipelines.PreProcessRequest.9.1.config"),
+                    appIncludeDirectory.CombineFile("Sitecore.uCommerce.Pipelines.PreProcessRequest.config"),
                     true,
                     () => versionChecker.IsEqualOrGreaterThan(new Version(9, 1)),
                     _loggingService),
-                new MoveFile(new FileInfo(Path.Combine(configIncludeDirectory.FullName, "Sitecore.uCommerce.Settings.config")),
-                    new FileInfo(Path.Combine(appIncludeDirectory.FullName, "Sitecore.uCommerce.Settings.config")),
+                new MoveFile(sitecoreDirectory.CombineFile("Sitecore.uCommerce.Settings.config"),
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/ComposeMoveSitecoreConfigIncludes.cs]: Should prolly be configIncludeDirectory

**Comment by Skidaddle99** [comment]
> 

I have tested a step yes

File: src/UCommerce.SiteCore.Installer/Steps/InstallDatabaseUcommerce.cs [hunk 6]
```diff
@@ -10,8 +11,7 @@ public class InstallDatabaseUcommerce : IStep
 
         public InstallDatabaseUcommerce(DirectoryInfo packageBasePath, InstallationConnectionStringLocator connectionStringLocator, IInstallerLoggingService logging)
         {
-            var migrationsDirectory =
-                new DirectoryInfo(Path.Combine(packageBasePath.FullName, "package", "files", "sitecore modules", "Shell", "Ucommerce", "Install"));
+            var migrationsDirectory =packageBasePath.CombineDirectory( "package", "files", "sitecore modules", "Shell", "Ucommerce", "Install");
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/InstallDatabaseUcommerce.cs:14]:
```suggestion
            var migrationsDirectory = packageBasePath.CombineDirectory( "package", "files", "sitecore modules", "Shell", "Ucommerce", "Install");
```

File: src/UCommerce.SiteCore.Installer/Steps/InstallStep.cs [hunk 7]
```diff
@@ -14,70 +15,95 @@ public class InstallStep : AggregateStep
...
+
                 new SeperateConfigSectionInNewFile("configuration/sitecore/settings",
-                    new FileInfo(Path.Combine(sitecoreDirectory.FullName, "web.config")),
-                    new FileInfo(Path.Combine(sitecoreDirectory.FullName, "App_Config", "Include", ".Sitecore.Settings.config")),
+                    sitecoreDirectory.CombineFile("web.config"),
+                    sitecoreDirectory.CombineFile("App_Config", "Include", ".Sitecore.Settings.config"),
                     loggingService),
-                new MoveDirectory(new DirectoryInfo(Path.Combine(sitecoreDirectory.FullName, "sitecore modules", "shell", "ucommerce", "install", "binaries")),
-                    new DirectoryInfo(Path.Combine(sitecoreDirectory.FullName, "bin", "uCommerce")),
+
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/InstallStep.cs:39]: Should we remove all these empty lines again?

**Review by estei — CHANGES_REQUESTED** [review]
A little bit to work on as I continue

File: src/UCommerce.SiteCore.Installer/Steps/SearchProviderCleanup.cs [hunk 8]
```diff
@@ -1,17 +1,18 @@
...
+using Ucommerce.Sitecore.Installer.FileExtensions;
 
 namespace Ucommerce.Sitecore.Installer.Steps
 {
     public class SearchProviderCleanup : AggregateStep
     {
         private readonly IInstallerLoggingService _loggingService;
 
-        public SearchProviderCleanup(string appsPath, IInstallerLoggingService loggingService)
+        public SearchProviderCleanup(DirectoryInfo appsPath, IInstallerLoggingService loggingService)
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/SearchProviderCleanup.cs:11]: Good catch on the type
Should we rename the variable alos?

File: src/UCommerce.SiteCore.Installer/Steps/SitecoreWebconfigMerger.cs [hunk 9]
```diff
@@ -19,22 +20,17 @@ public SitecoreWebconfigMerger(DirectoryInfo sitecoreDirectory, IInstallerLoggin
         public async Task Run()
         {
             _loggingService.Information<SitecoreWebconfigMerger>("Merging Sitecore and Ucommerce config files...");
-            var ucommerceInstallPath = Path.Combine("sitecore modules", "Shell", "ucommerce", "install");
-            var mergeConfig = new MergeConfig(new FileInfo(Path.Combine(_sitecoreDirectory.FullName, "web.config")),
+            var ucommerceInstallPath = _sitecoreDirectory.CombineDirectory("sitecore modules", "Shell", "ucommerce", "install");
```
Comment (estei) [inline src/UCommerce.SiteCore.Installer/Steps/SitecoreWebconfigMerger.cs:23]: Should we rename this to directory?

**Review by estei — CHANGES_REQUESTED** [review]
A couple of renames when going from string to directoryinfo

**Review by Victor-Allan — APPROVED** [review]

**Review by estei — APPROVED** [review]
GJ

**Pull request merged**